From f1a9fc2fafcd4c34b9f77dad7b6418b3bdf47dc0 Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Sat, 16 Feb 2008 19:28:24 +0200
Subject: [PATCH]  Cygwin compilation fixes

Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 afio.c     |    5 +++++
 afio.h     |    2 +-
 compfile.c |    2 +-
 match.c    |    2 +-
 4 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/afio.c b/afio.c
index f2f7741..f6b0fd6 100644
--- a/afio.c
+++ b/afio.c
@@ -160,6 +160,11 @@ static char *ident = "$Header: /u/buhrt/src/afio/RCS/afio.c,v 2.3 1991/09/25 20:
 #include <stdio.h>
 #include <errno.h>
 
+#ifdef __CYGWIN__
+#include <utime.h>
+#include <sys/wait.h>
+#endif
+
 #ifdef sun
 #include <sys/types.h>
 #include <utime.h>
diff --git a/afio.h b/afio.h
index fecd575..808ef5f 100644
--- a/afio.h
+++ b/afio.h
@@ -477,7 +477,7 @@ VOIDFN readcheck (char **);
 #ifndef MKDIR
 int rmdir (char *);
 #endif
-#if !defined (linux) && !defined(__FreeBSD__) && !defined(sun)
+#if !defined (linux) && !defined(__FreeBSD__) && !defined(sun) && !defined(__CYGWIN__)
 VOIDFN (*signal ())();
 #endif
      int fswrite (int, char*, uint);
diff --git a/compfile.c b/compfile.c
index bf167fe..e433c48 100644
--- a/compfile.c
+++ b/compfile.c
@@ -210,7 +210,7 @@ void waitforgzip()
  * version;
  */
 
-#if ( defined(sun) && defined(__svr4__) )
+#if ( defined(sun) && defined(__svr4__) ) || defined(__CYGWIN__)
 #include <dirent.h>
 #else
 #include <sys/dir.h>
diff --git a/match.c b/match.c
index 6e53648..3160343 100644
--- a/match.c
+++ b/match.c
@@ -146,7 +146,7 @@ nameaddfile(fname,ptype)
  *
  * Compare a pathname with the pattern list. Returns 0 for
  * a match, -1 otherwise.
- * 
+ *
  * This new version uses the GCC library function fnmatch()
  * to provide real pattern matching.
  *
-- 
1.5.3.8

